v19.2/experimental-audit.md

```diff
@@ -88,6 +88,26 @@ If your deployment requires particular lifecycle and access policies for audit l
 
 {% include {{ page.version.version }}/misc/schema-change-view-job.md %}
 
+## Performance considerations
+
+Enabling SQL audit logs will impact performance, as every query that causes a logging event must access the disk of the node on which audit logging is enabled.
```

Comment:
I think this text could do a better job of explaining there are different user stories, and that one should be careful to choose the features they need:

- from a security perspective, an "audit log" provides what is known as *non-repudiability*, the ability for an operator to know what users (especially malicious users) did with the confidence that *all* of what the user did is in the log. 

  Non-repudiability requires the writes to disk to be synchronous, because otherwise a malicious user could do something bad then immediately crash the node and then the evidence would be erased (in that case, the log is repudiable and cannot be used for audit purposes)

- separately, users may want/need a trace of all SQL statements executed, so that they can more readily troubleshoot application bugs or performance problems. These logs typically do not need to be complete, let alone non-repudiable. For that use case, the "audit" logs are overkill and detrimental (because of their performance impact).

For the security story, our current logging story is not exactly complete (we have more work to do).

For the SQL analysis / troubleshooting story, our current logging story has a bit _too many_ features -- in particular it's tempting to include the "experimental audit logging" as one of the logging systems that can be used for it, even though it was not designed for it.


Now I realize after writing all this that this general way to look at things is not the one that has been adopted in our doc around logging (yet). This may be a sign that a larger rework is in order. I do not know what you think would be the best first next step here.

Reply:
Excellent feedback, @knz. 

I think the best first step would be to:

- Go into a little more detail about the non-repudiation guarantee on the [`EXPERIMENTAL AUDIT` reference page](https://www.cockroachlabs.com/docs/stable/experimental-audit).

- Call out audit logging as a security feature (or as a logging feature for security), rather than a logging feature for troubleshooting in as many places as possible. 

I've added a commit that does both of these things, in the v19.2 docs (I will propagate these changes to v20.1 and v20.2 docs once I get your approval).

Note that the [SQL Audit Logging page](https://www.cockroachlabs.com/docs/stable/sql-audit-logging.html) is already listed as a [security feature in our security docs](https://www.cockroachlabs.com/docs/stable/security-overview.html#security-features-in-cockroachdb).

Reply:
@knz Ping on re-review (but no rush). :)